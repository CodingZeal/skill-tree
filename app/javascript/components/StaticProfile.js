import React from "react";

function StaticProfile({ email, firstname, lastname, timezone }) {
  return (
    <div className="staticprofile">
      <h1 className="card-header">{`${firstname}'s Profile`}</h1>
      <div className="card">
        <h1 id="fullname">
          {firstname} {lastname}
        </h1>
        <h2 className="card-info" id="email">
          <span aria-label="envelope" role="img">
            {" "}
            ‚úâÔ∏è
          </span>{" "}
          {email}
        </h2>
        <h2 className="card-info" id="timezone">
          <span aria-label="globe" role="img">
            {" "}
            üåê
          </span>{" "}
          {timezone}
        </h2>
      </div>
    </div>
  );
}

export default StaticProfile;
