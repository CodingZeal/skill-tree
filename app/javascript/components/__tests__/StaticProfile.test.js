import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";

import StaticProfile from "../StaticProfile";

Enzyme.configure({ adapter: new Adapter() });

describe("<StaticProfile>", () => {
  const profile = {
    email: "john@example.com",
    firstname: "John",
    lastname: "Doe",
    timezone: "UTC",
    url: "iou22o38f4"
  };
  const wrapper = shallow(<StaticProfile profile={profile} />);
  it("renders corretly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("displays user fullname", () => {
    expect(wrapper.find("#fullname").length).toEqual(1);
  });
  it("displays user email", () => {
    expect(wrapper.find("#email").length).toEqual(1);
  });
  it("displays user timezone", () => {
    expect(wrapper.find("#timezone").length).toEqual(1);
  });
});
