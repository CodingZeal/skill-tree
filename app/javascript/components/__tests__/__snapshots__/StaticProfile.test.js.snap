// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaticProfile> renders corretly 1`] = `
<div
  className="staticprofile"
>
  <h1
    className="card-header"
  >
    undefined's Profile
  </h1>
  <div
    className="card"
  >
    <h1
      id="fullname"
    >

    </h1>
    <h2
      className="card-info"
      id="email"
    >
      <span
        aria-label="envelope"
        role="img"
      >

        ✉️
      </span>

    </h2>
    <h2
      className="card-info"
      id="timezone"
    >
      <span
        aria-label="globe"
        role="img"
      >

        🌐
      </span>

    </h2>
  </div>
</div>
`;
